src/main/java/org/ebyhr/trino/storage/StorageMetadata.java

```diff
@@ -167,15 +154,69 @@ private ConnectorTableMetadata getStorageTableMetadata(ConnectorSession session,
             return null;
         }
 
-        return new ConnectorTableMetadata(tableName, table.getColumnsMetadata());
+        return new ConnectorTableMetadata(tableName.toSchemaTableName(), table.getColumnsMetadata());
+    }
+
+    /**
+     * Simplified variant of {@link SchemaTableName} that doesn't case-fold.
+     */
+    private static class SchemaTablePair
```

Comment:
Name: That's fair, the intention here was to communicate "always use this instead of `SchemaTableName`, except as required to communicate with the Trino API", rather than focus on the current difference.

Record: Sure, didn't want to use it since I couldn't see any current usage of it and wasn't sure about how it'd interact with whatever JRE compatibility policy you have.

Reply:
Record: 4e734af4ba445f60b84650f11ba4a22476167aa5

Reply:
Name: e8a9a139ac2220edfdce7df6c7ca600a77573440